planner/core/logical_plan_builder.go

```diff
@@ -1206,6 +1206,9 @@ func (a *havingWindowAndOrderbyExprResolver) Enter(n ast.Node) (node ast.Node, s
 		a.inAggFunc = true
 	case *ast.WindowFuncExpr:
 		a.inWindowFunc = true
+		if a.curClause == unknowClause {
```

Comment:
Why not just use `inWindowFunc` to determine the resolve rule?

Reply:
We can use `inWindowFunc` here. But I'm not sure how windowClause is used and what's the difference between them.

Reply:
@lamxTyler Take `select a, b as z from t order by (sum(z) over())` for example,
If we use `inWindowFunc`, resolveFieldsFirst will be false, which should be true.

Reply:
Actually `windowClause` is only readed in one place, and from the code itself, it is set at the same time with `inWindowFunc`. You can search the code to find these facts.

Reply:
> Actually `windowClause` is only readed in one place, and from the code itself, it is set at the same time with `inWindowFunc`. You can search the code to find these facts.

I use fieldList to replace windowClause, and I think it's more understandable. PTAL

Reply:
> @lamxTyler Take `select a, b as z from t order by (sum(z) over())` for example,
> If we use `inWindowFunc`, resolveFieldsFirst will be false, which should be true.

But currently it is false? Since `inWindowFunc` is true.

Reply:
> > @lamxTyler Take `select a, b as z from t order by (sum(z) over())` for example,
> > If we use `inWindowFunc`, resolveFieldsFirst will be false, which should be true.
> 
> But currently it is false? Since `inWindowFunc` is true.

Sorry for my mistake, when using `inWindowFunc`, we need also deal with `if index == -1 && a.curClause != windowClause {` . So we can't just use `inWindowFunc` and leave alone `windowClause`. The concept `windowClause` is too cunfusing to distinguish from `inWindowFunc`. In fact, it should be `fieldList`. Base on this, the root cause of this bug is obvious, for some `column` in `filedList` but not in `WindowFunc`, we should set `resolveFieldsFirst` as `false`